### ISSUE TITLE
Unit tests

### ISSUE THREAD

**Issue by mjkoo** [polyswarm/relay#18]
**Is your feature request related to a problem? Please describe.**
We've deferred fleshing out our unit test suite due to API changes and difficulty testing interconnected components in isolation, but it's past time we improve this.

In addition to improving coverage, we should
- [x] Include unit tests in CI
- [ ] Report coverage from gitlab CI pipeline runs in badge
- [ ] Run clippy lints in CI
- [x] Ensure cargo fmt clean in CI https://github.com/rust-lang-nursery/rustfmt#checking-style-on-a-ci-server

**Comment by mjkoo** [comment]
Whats our coverage at and where are the new tests, I dont see a PR

**Comment by myhndl** [comment]
We can't close this ticket until coverage is in CI and badges are in README

**Comment by myhndl** [comment]
@zv I saw a pull request or something on this today - where are we on this?

**Comment by myhndl** [comment]
blocked by: https://github.com/polyswarm/relay/issues/13

**Comment by ceseale** [comment]
@myhndl I don't think this blocked anymore https://github.com/polyswarm/relay/pull/31 fixes the test.

**Comment by myhndl** [comment]
Oh okay great @ceseale !

@zv please take note.
